### PR TITLE
TripleDiff pull request

### LINKED ISSUE #9: TripleDiff: compute diff for two given sets of Statement instances
Function [computeDiffForTwoTripleSets](https://github.com/k00ni/knorke/blob/9b48a457362b99c75a11f268773e8d5a6345fc26/src/Knorke/TripleDiff.php#L88)

**Features**:
* [ ] compute the diff of two sets containing Statement instances (triples). return two separate arrays each containing the Statement instances which are unique to the according set.
* [ ] ignore for doublings
* [ ] edge cases

Ping: @GeorgesAlkhouri

**Assumed time required**: 3 hours

### PR DESCRIPTION
**Pull request by GeorgesAlkhouri**
Implementing method to compare statement instances by computing hashes. Therefore added a new statement class: HashableStatement.
Also, graph URIs can be considered for hash computation but are disabled by default.
Close #9 #10

### CODE REVIEW

**Comment by k00ni** [comment]
There are conflicts that must be resolved. Could you check please and resolve, if they are not that big of a deal?

**Comment by GeorgesAlkhouri** [comment]
I think this is ready to be merged.

**Comment by k00ni** [comment]
Thanks for the feedback, will look into that next week.

File: tests/Knorke/TripleDiffTest.php [hunk 1]
```diff
@@ -170,7 +223,7 @@ public function testComputeDiffForTwoTripleSets()
     public function testComputeDiffOneSetEmpty()
     {
         list($set1, $set2ToBeIgnored) = $this->generate2TestSets();
-
+        // print_r($set1[0]);
```
Comment (k00ni) [inline tests/Knorke/TripleDiffTest.php]: Remove such lines please.

File: tests/Knorke/TripleDiffTest.php [hunk 2]
```diff
@@ -194,4 +247,86 @@ public function testComputeDiffOneSetEmpty()
...
+                    $this->nodeFactory->createNamedNode('http://set2/5'),
+                    $this->nodeFactory->createLiteral('666'),
+                    $this->nodeFactory->createNamedNode('http://graph2/')
+                )
+              )
+          ),
+          $diffArray
+      );
+    }
+    public function testDiffWithOneEmptyQuadSet()
```
Comment (k00ni) [inline tests/Knorke/TripleDiffTest.php]: New line between these functions please (like the ones before).

File: tests/Knorke/TripleDiffTest.php [hunk 3]
```diff
@@ -194,4 +247,86 @@ public function testComputeDiffOneSetEmpty()
...
+                  $this->nodeFactory->createNamedNode('http://graph1/')
+              )
+            ),
+            array()
+        ),
+        $diffArray
+      );
+    }
+
+    // public function testWithBlankNodes()
```
Comment (k00ni) [inline tests/Knorke/TripleDiffTest.php]: Remove these for now please

**Comment by k00ni** [comment]
I finally found the time to check your code. 

Your contribution overall looks good. But i found it very problematic, that you require the use of a new class+interface HashableStatement/HashableStatementImpl, because nothing in the Saft framework is able to generate these. I have to transform a given set of Statement instances into HashableStatement instances before using them, which is very inefficient. Saft's AbstractStatement already provides a [function](https://github.com/SaftIng/Saft/blob/d4f8e0192a23616c3e225a7eeafe6bbf3635b576/src/Saft/Rdf/AbstractStatement.php#L51-L67) to get a string representation of a given Statement. Please move the hashing-part into the TripleDiff class and reuse the mentioned function. If it is not sufficient, lets discuss possible adaptions in Saft itself.

Extending the RDF foundation of Saft is very critical and needs further discussion in general. Knorke has to be like an add-on, if changes on the RDF-basement are required, they have to be made in Saft itself.

**Comment by GeorgesAlkhouri** [comment]
Yes you are right,  i'll fix this.

File: src/Knorke/TripleDiff.php [hunk 4]
```diff
@@ -52,49 +53,88 @@ public function __construct(
...
+
+        $statementSet1 = array_map($resultsToStatements, iterator_to_array($graphResult1));
+        $statementSet2 = array_map($resultsToStatements, iterator_to_array($graphResult2));
+
+        return $this->computeDiffForTwoTripleSets($statementSet1, $statementSet2);
     }
 
     /**
-     * Computes the diff of two sets containing triples (Statement instances).
+     * Computes the diff of two sets containing triples or quads (HashableStatement instances).
```
Comment (k00ni) [inline src/Knorke/TripleDiff.php]: Could you replace HashableStatement with Statement here please?

**Pull request merged**